doc/quickstart.rst

```diff
@@ -0,0 +1,178 @@
+.. _quickstart:
+
+Quickstart
+==========
+
+The following shows how to get started with pyOptSparse by solving
+Schittkowski’s TP37 constrained problem. First, we show the complete
+program listing and then go through each statement line by line:
+
+.. literalinclude:: ../examples/tp037.py
+   :start-after: # rst begin import
+   :end-before: # rst end opt
+
+Start by importing the pyOptSparse package:
+
+.. literalinclude:: ../examples/tp037.py
+   :start-after: # rst begin import
+   :end-before: # rst begin objfunc
+
+Next we define the objective function that takes in the design
+variable *dictionary* and returns a *dictionary* containing the
+constraints and objective, as well as a (boolean) flag indicating if
+the objective function evaluation was successful. For the TP37, the
+objective function is a simple analytic function:
+
+.. literalinclude:: ../examples/tp037.py
+   :start-after: # rst begin objfunc
+   :end-before: # rst begin optProb
+
+Notes:
+
+  1. The ``xdict`` variable is a dictionary whose keys are the names
+     from each ``addVar()`` and ``addVarGroup()`` call. The line::
+
+       x = xdict['xvars']
+
+     retrieves an array of length 3 which are all the variables for
+     this optimization. 
+
+  2. The line::
+    
+       conval = [0]*2
+
+     creates a list of length 2, which stores the numerical values of
+     the two constraints. The ``funcs`` dictionary return must contain
+     keys that match the constraint names from ``addCon`` and
+     ``addConGroup``  as well as the objectives from ``addObj`` calls. This 
+     is done in the following calls::
+
+       funcs['obj'] = -x[0]*x[1]*x[2]
+       funcs['con'] = conval
+
+Now the optimization problem can be initialized:
+
+.. literalinclude:: ../examples/tp037.py
+   :start-after: # rst begin optProb
+   :end-before: # rst begin addVar
+
+This creates an instance of the optimization class with a name and a
+reference to the objective function. To complete the setup of the
+optimization problem, the design variables and constraints need to be defined. 
+
+Design variables and constraints can be added either one-by-one or as
+a group. Adding variables by group is generally recommended for
+related variables:
```

Comment:
maybe add a quick link to `addVarGroup`? If you don't think it creates confusion

Reply:
Do you mean a link to the API documentation?

Reply:
Yes!

Reply:
I would prefer to do that with the `guide.rst`, but not link anything in `quickstart` since it's meant to be...well quick :P

Reply:
I added links to existing mentions of functions, but did not add any additional links